src/harness.js

```diff
@@ -187,13 +194,20 @@ class Harness {
   }
 
   /**
-   * Set the current mock-time of the harness
+   * Set the current mock-time of the harness. Mocks global time {@link https://sinonjs.org/releases/v1.17.6/fake-timers/|uses sinon}
    * @param {Date|number|string} now A Date object or a value which can be parsed into a Date
+   * 
    */
   setNow(now) {
+    if (!now) {
+      throw Error('undefined date passed to setNow');
+    }
+
     const parseableNow = typeof now === 'object' ? now.getTime() : now;
-    this._now = toDate(parseableNow).getTime();
-    return this.page.evaluate(innerNow => window.now = new Date(innerNow), this._now);
+    const asTimestamp = toDate(parseableNow).getTime();
```

Comment:
Not really relevant but couldn't we avoid `asTimestamp` variable here and just pass `toDate(parseableNow).getTime()` directly to `this._now` ?

Reply:
Wouldn't this mean redoing the computation again in L209? Variables help by storing the result for reuse